bootstrapping/bootstrapping.py

```diff
@@ -0,0 +1,222 @@
+from typing import Sequence, Optional, Iterator
+from collections import Counter
+import copy
+
+from evidence_modeling.frequency import Frequency
+from bootstrapping.contexts import ContextType, Context
+from gender import Gender
+
+ALLOWED_CONTEXT_MODELS = [ContextType.LEFT_WHOLE_WORD, ContextType.RIGHT_WHOLE_WORD, ContextType.BILATERAL_WHOLE_WORD,
+                          ContextType.LEFT_SUFFIX, ContextType.RIGHT_SUFFIX, ContextType.BILATERAL_SUFFIX]
+
+
+def is_context_gender_specific(context_frequency: Frequency, fraction_to_allow: float = 0.5) -> Optional[Gender]:
+    """
+    Defines the condition for filtering, whether the context absolute counts are relevant enough to decide whether the
+    context is masculine or feminine.
+    :param fraction_to_allow:
+    :param context_frequency: Frequency containing absolute counts of the context.
+    :return: Specific gender if the counts are relevant enough, or None if we cannot decide.
+    """
+
+    # TODO: here are multiple possibilities on the conditions:
+    # 1. fem > masc + quest/2, resp. masc > fem + quest/2
+    # 2. fem > 0 and masc = 0 (or fem > n and masc = 0)
+    # 3. fem > masc
```

Comment:
2 and 3 are not in the order below.

Reply:
In general, it is not a good practice to leave commented-out code in the final version. It would be better to create three separate functions (e.g., `...simple`, `...strict`, ...; each with a simple comment). Then, you can have a parameter (or constant) defining which one to use.

Reply:
fixed